AlphaWalletShare/ShareViewController.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  ShareViewController.swift
+//  AlphaWalletShare
+//
+//  Created by Vladyslav Shepitko on 10.11.2020.
+//
+
+import UIKit
+
+@objc(ShareViewController)
+class ShareViewController: UIViewController {
+    
+    private let storage = SharedDefaults()
+
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        enum AnyError: Error {
+            case canceled
+        }
+
+        guard let context = extensionContext else {
+            return
+        }
+
+        guard let extensionItem = context.inputItems.first as? NSExtensionItem else {
+            context.cancelRequest(withError: AnyError.canceled)
+            return
+        }
+
+        let valueResolver = DefaultItemProviderValueResolver()
+        extensionItem.resolveAttachments(valueResolver: valueResolver) { attachment in
+            if let attachment = attachment {
+                self.showOpenURL { result in
+                    switch result {
+                    case .success:
+                        self.storage.attachement = attachment
```

Comment:
Do you think we can encode the URL to open directly into the custom URL, then we slip the temporary storage? eg. `awallet://share_content?u=example.com` (with % encoding)